.evergreen/find-python3.sh

```diff
@@ -198,6 +198,8 @@ find_python3() (
   local -a bins=()
   local bin=""
 
+  trap "echo 'find_python3 failed!'" EXIT
+
```

Comment:
```suggestion
```
I strongly discourage the use of `trap`.

Given `find-python3.sh` is designed to be sourced, this trap will be unconditionally set in the scope of the parent script sourcing this file. That means this trap command will only be executed when the _parent_ script exits (or whatever script(s) may be sourcing the parent, etc.), which is not the intent of this command. Furthermore, it may also silently override the caller's own trap commands. This script was deliberately designed to avoid any such "leaks" that may affect the user's scripting environment.

I recommend simply adding `echo` guards around calls to the `find_python3` function in files such as `start-orchestration.sh` or `activate-*venv.sh` rather than modifying `find-python3.sh`. It may be a bit repetitive/tedious but it is a far safer and simpler-to-maintain approach.